client/src/client.rs

```diff
@@ -1105,6 +1147,95 @@ pub mod events_api {
         }
     }
 
+    /// Iterator for getting events from the `WebSocket` stream.
+    #[derive(Debug)]
+    pub struct AsyncEventIterator {
+        stream: AsyncWebSocketStream,
+        handler: flow::Events,
+    }
+
+    impl AsyncEventIterator {
+        /// Constructs `AsyncEventIterator` and sends the subscription request.
+        ///
+        /// # Errors
+        /// Fails if connecting and sending subscription to web socket fails
+        pub async fn new(handler: flow::Init) -> Result<Self> {
+            let InitData {
+                first_message,
+                req,
+                next: handler,
+            } = FlowInit::<http_default::DefaultWebSocketRequestBuilder>::init(handler);
+
+            let mut stream = req.build()?.connect_async().await?;
+            stream.send(WebSocketMessage::Binary(first_message)).await?;
+
+            let handler = loop {
+                let mes = stream
+                    .next()
+                    .await
+                    .ok_or(eyre!("Expect to receive message"))?;
+                debug!(?mes, "Received message");
+                match mes {
+                    Ok(WebSocketMessage::Binary(message)) => break handler.message(message)?,
+                    Ok(_) => continue,
+                    Err(WebSocketError::ConnectionClosed | WebSocketError::AlreadyClosed) => {
+                        return Err(eyre!("WebSocket connection closed."))
+                    }
+                    Err(err) => return Err(err.into()),
+                }
+            };
+            debug!("`EventIterator` created successfully");
+            Ok(Self { stream, handler })
+        }
+
+        /// Return next event
+        pub async fn next(&mut self) -> Option<Result<Event>> {
```

Comment:
It seems to me it would be great to implement a more general [Stream](https://docs.rs/tokio/latest/tokio/stream/index.html) instead of a specific Iterator.

Reply:
I implemented `Stream` for `AsyncEventStream` as you suggested.